src/js/components/Button/Button.js

```diff
@@ -92,12 +96,80 @@ const getIconColor = (paths = [], theme, colorProp, kind) => {
   return result[1] || undefined;
 };
 
+const getBadgeDimension = (dimension, badgeProp, badgeRef, theme) => {
```

Comment:
There might be enough code here to warrant a separate Button/Badge.js file.

Reply:
I think I like the idea of that, will try it out locally and see if it feels more readable.